src/api_server/src/lib.rs

```diff
@@ -72,14 +73,14 @@ pub struct ApiServer {
 impl ApiServer {
     pub fn new(
         mmds_info: Arc<Mutex<Mmds>>,
-        vmm_shared_info: Arc<RwLock<InstanceInfo>>,
+        instance_info: RefCell<InstanceInfo>,
```

Comment:
You can avoid the need for `RefCell` if you make `handle_request()` and `serve_vmm_action_request()` take _mutable_ reference of `self`.

Example of that can be see in https://github.com/firecracker-microvm/firecracker/pull/2282 which is another pending PR which needs `&mut self` in those functions.

Reply:
Yes thanks! Fixed.